epochStart/shardchain/peerMiniBlocksSyncer_test.go

```diff
@@ -344,3 +345,458 @@ func TestValidatorInfoProcessor_ProcesStartOfEpochWithMissinPeerMiniblocksTimeou
 
 	require.Equal(t, process.ErrTimeIsOut, processError)
 }
+
+func TestValidatorInfoProcessor_SyncValidatorsInfo(t *testing.T) {
+	t.Parallel()
+
+	t.Run("sync validators info with nil block body", func(t *testing.T) {
+		t.Parallel()
+
+		args := createDefaultArguments()
+		syncer, _ := NewPeerMiniBlockSyncer(args)
+
+		missingValidatorsInfoHashes, validatorsInfo, err := syncer.SyncValidatorsInfo(nil)
+		assert.Nil(t, missingValidatorsInfoHashes)
+		assert.Nil(t, validatorsInfo)
+		assert.Equal(t, epochStart.ErrNilBlockBody, err)
+	})
+
+	t.Run("sync validators info with missing validators info", func(t *testing.T) {
+		t.Parallel()
+
+		args := createDefaultArguments()
+		args.ValidatorsInfoPool = &testscommon.ShardedDataStub{
+			SearchFirstDataCalled: func(key []byte) (value interface{}, ok bool) {
+				return nil, false
+			},
+		}
+		syncer, _ := NewPeerMiniBlockSyncer(args)
+
+		body := &block.Body{
+			MiniBlocks: []*block.MiniBlock{
```

Comment:
also here, either use the aforementioned function or create a new one

Reply:
Done